Add Meteoformer for meteorological forecasting

zhangpu00000 opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleScience/pull/96 -->
### PR types
Others

### PR changes
Others

### Describe
add Meteoformer model for meteorological forecasting
add docs for Meteoformer 
add examples for Meteoformer 
add era5meteo_dataset for Meteoformer

zhangpu00000 committed:
Add Meteoformer for meteorological forecasting

HydrogenSulfate commented:
@zhangpu00000 感谢提交PR，请使用pre-commit对代码进行格式化
![image](https://github.com/user-attachments/assets/d5d01e27-6655-46a2-86e6-9c025ae274ce)

zhangpu00000 committed:
pre-commit

使用pre-commit对代码格式化。

HydrogenSulfate commented on examples/meteoformer/conf/train.yaml:
meteoformer.yaml

HydrogenSulfate commented on examples/meteoformer/main.py:
```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
```suggestion
# Copyright (c) 2025 PaddlePaddle Authors. All Rights Reserved.
```

HydrogenSulfate commented on examples/meteoformer/main.py:
```diff
@@ -0,0 +1,149 @@
+    solver.eval()
+
+
+@hydra.main(version_base=None, config_path="./conf", config_name="train.yaml")
```
```suggestion
@hydra.main(version_base=None, config_path="./conf", config_name="meteoformer.yaml")
```

HydrogenSulfate commented on ppsci/arch/meteoformer.py:
```diff
@@ -0,0 +1,435 @@
+        hid_T: int = 256,
+        N_S: int = 4,
+        N_T: int = 4,
+        incep_ker: Tuple[int, ...] = [3, 5, 7, 11],
```
默认参数不建议使用list这种可变类型，而是tuple
```suggestion
        incep_ker: Tuple[int, ...] = (3, 5, 7, 11),
```

HydrogenSulfate commented on ppsci/arch/meteoformer.py:
```diff
@@ -0,0 +1,435 @@
+        groups: int = 8,
+        num_classes: int = 4,
+    ):
+        super(Meteoformer, self).__init__()
```
```suggestion
        super().__init__()
```

HydrogenSulfate reviewed (commented):
感谢提交PR，代码没有大问题，几处细节还麻烦修改一下

HydrogenSulfate commented on docs/zh/examples/meteoformer.md:
````diff
@@ -0,0 +1,189 @@
+
+该模块使用两层Transformer，提取空间特征更新节点特征：
+
+``` py linenums="8" title="ppsci/arch/Meteoformer.py"
````
本文档的linenums应该跟下方的代码引用起始位置一致，请全文检查一遍

zhangpu00000 committed:
Merge remote-tracking branch 'upstream/develop' into develop

zhangpu00000 committed:
Changes in Meteoformer for meteorological forecasting

zhangpu00000 committed:
Add Climateformer for climate prediction

add Climateformer model for climate prediction
add docs for Climateformer
add examples for Climateformer
add era5climate_dataset for Climateformer

zhangpu00000 committed:
Merge branch 'develop' into climateformer

zhangpu00000 committed:
Changes for Climateformer

Changes for Climateformer

zhangpu00000 committed:
Revised Version of Climateformer for climate prediction

Revised Version of Climateformer for climate prediction

zhangpu00000 committed:
Update __init__.py

zhangpu00000 committed:
Update __init__.py

zhangpu00000 committed:
Revised Version of Meteoformer

zhangpu00000 committed:
Merge branch 'develop' into develop

zhangpu00000 committed:
Delete climateformer.md

zhangpu00000 committed:
Update meteoformer.md

zhangpu00000 committed:
Update meteoformer.md

HydrogenSulfate commented on ppsci/data/dataset/era5meteo_dataset.py:
```diff
@@ -0,0 +1,184 @@
+
+import numpy as np
+import paddle
+import xarray as xr
```
跟上一个PR类似，使用try对xarray进行处理

zhangpu00000 committed:
Update era5meteo_dataset.py

zhangpu00000 committed:
Update era5meteo_dataset.py

zhangpu00000 committed:
Update meteoformer.yaml

HydrogenSulfate commented on examples/meteoformer/conf/meteoformer.yaml:
```diff
@@ -0,0 +1,76 @@
+  compute_metric_by_batch: true
+  eval_with_no_grad: true
+  batch_size: 8 # 16
+
```
```suggestion
```

HydrogenSulfate committed:
Update examples/meteoformer/conf/meteoformer.yaml

zhangpu00000 committed:
Merge branch 'develop' into develop

zhangpu00000 committed:
Update meteoformer.yaml

zhangpu00000 committed:
Merge branch 'develop' of https://github.com/zhangpu00000/PaddleScience into develop

HydrogenSulfate commented:
@zhangpu00000 顺便解决一下冲突

zhangpu00000 committed:
Merge branch 'develop' into develop

HydrogenSulfate commented:
@zhangpu00000 
<img width="898" height="467" alt="image" src="https://github.com/user-attachments/assets/8f61ccf9-0fa4-4e7e-86c4-f04fe9e9fcac" />
还是有问题的，用pre-commit过一下

HydrogenSulfate committed:
Update __init__.py

refine import order

zhangpu00000 committed:
Merge branch 'develop' into develop

HydrogenSulfate reviewed (approved):
LGTM
